File: C/Algorithms/Sorting/Bubble_Sort.c
#include <stdio.h>
#include <stdlib.h>

/* 
@breif This function is used to prints the values of the array
@param *arr -> pointer of the array to be printed
@param array_length -> lenght of the array to be printed form 0 to array_length
@return void
*/

void print_arr(int *arr, int array_length){
    for(int i=0; i<array_length; i++){
        printf("%d, ",arr[i]);
    }
}


/* 
@breif This function implements BUBBLE SORT it is used to sort the array in ascending order 
@param *arr -> pointer of the array to be sorted
@param array_length -> lenght of the array
@return void
*/

void bubble_sort(int *arr, int array_length) {
    // Outer loop to iterate over each element of the array
    for (int i = array_length - 1; i >= 1; i--) {
        // Inner loop to perform comparisons and swaps
        for (int j = 0; j < i; j++) {
            // Compare adjacent elements
            if (arr[j] > arr[j + 1]) {
                // Swap the elements if they are in the wrong order
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}



int main(void){
    int arr[] = {4,3,5,87,2,7,9,10};
    int length = sizeof(arr)/sizeof(int);
    insertion_sort(arr, length);
    print_arr(arr,length);    

    return 0;
}

File: Go/Sorting/BubbleSort.go
package main

import (
	"fmt"
)

func Bubblesort(n int, arr []int) {

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if arr[i] > arr[j] {
				arr[i], arr[j] = arr[j], arr[i]
			}
		}
	}

}

func main() {

	arr := [6]int{3, 5, 1, 10, 6, 2}
	n := 6 // its the length of the array

	Bubblesort(n, arr[:])

	fmt.Print("The sorted array is printed as")

	for i := 0; i < n; i++ {
		fmt.Print(" ", arr[i])
	}

}
